Added reduced client functionality

yokujin opened the pull request:
I had to use your daemon in Python 3 only environment. And it is known for original daemon not to work with Python 3.
So I had to implement a subset of client commands to control daemon. I implemented `start`, `stop` and `status` commands.

If anyone interested in this here is PM )

freman commented:
Option to use unix socket instead of tcp socket?

Probably easiest to take address instead of host+port

ie

```
--address 127.0.0.1:9001
```

or

```
--address unix:///$home/supervisord/sock
```

yokujin commented:
Sounds good. I'll look into this. 
I've no idea why I switched to combination of host and port.

yokujin commented:
- Replaced server host and port combination into `--serverurl` parameter as original supervisord does.
- Implemented `shutdown` command. Without security at the moment. I'll work on that.
- Added ability to daemonize supervisor.

freman commented on config.go:
```diff
@@ -349,7 +349,11 @@ func (c *ConfigEntry) GetString(key string, defValue string) string {
 
 	if ok {
-		return s
+		env := NewStringExpression("here", c.ConfigDir)
```
Without stomping all over anyones toes, maybe use the new method I added in this patch?

https://github.com/ochinchina/supervisord/pull/29/files#diff-b4bda758a2aef091432646c354b4dc59R358
